docs/proposal-platforms.md

```diff
@@ -235,31 +235,183 @@ suite-run functionality:
+       Specifies the directory where work directories are created. If
+       specified the workflow work directory will be created in
+       `<work dir>/<workflow-name>/work` and a symbolic link will be
+       created from `$HOME/cylc-run/<workflow-name>/work`. If not
+       specified the workflow work directory will be created in
+       `$HOME/cylc-run/<workflow-name>/work`.
+   * What happens if you want to be able to have some workflows which use
+     different directory configurations on the same "platform"? For
+     example, you might have some workflows where you want the run
+     directory on $HOME and others on $DATADIR.
+     * For platforms other than localhost the answer is to create separate
+       platforms and then choose the appropriate platform in the workflow.
+     * For localhost it is more complicated since there is only one localhost
+       platform and the settings are applied earlier; at workflow install or
+       startup time (rather than prior to first job submission). We will need a
+       way to override the directory settings for localhost, e.g. on the command
+       line.
+       * Cylc 8 already has a way to
+         [symlink individual run directories](https://github.com/cylc/cylc-flow/pull/2935).
+         The interface may change but this proposal will still support this
+         functionality.
+   * What environment variables can be used in these settings?
+     * Cylc currently only allows `$HOME` or `$USER` in the
+       [run and work directory settings](https://cylc.github.io/doc/built-sphinx/appendices/site-user-config-ref.html#hosts-host-run-directory).
+       With Rose you can use any variable which is available when the
+       remote-init command is invoked on the remote platform. Should we adopt a
+       similar approach for Cylc?
+     * Rose also provides the option of using the login shell for remote
+       commands (which may give access to more environment variables). Should we
+       add similar support to Cylc for sites that allow it?
+   * Any platforms which use the same target must use the same symlink
+     directories.
+     * In order to reduce duplication (and configuration errors) we will support
+       platform inheritance so that you can define a platform based on an
+       existing platform.
+       * Note: `inherit` will imply using the same `install target`.
+     * Do we need comms method to be the same as well?
+   * The `[symlink dir]` settings are only applied when a workflow is installed
+     (or first run on a target). Therefore, changes to these settings have no
+     affect on running or restarting workflows.
+   * At the moment this proposal does not provide any way to relocate the
+     top level `$HOME/cylc-run` directory (which you can do currently via the
+     existing Cylc `run directory` setting). The assumption is that it is
+     sufficient to provide a way to move the run directory and there is no
+     reason not to use `$HOME/cylc-run`.
```

Comment:
Does this seem reasonable?

Reply:
Yes. 
Consider where it might cause a problem:
1. User has unrelated top level `cylc-run` dir: Just don't do this!
2. User keeps test/dev suites in a different run dir so that if something goes wrong they can delete it all (mia culpa!): Probably bad practice anyway.
3. ...